source/parser.hera

```diff
@@ -6536,18 +6536,22 @@ TypeSuffix
   # TypeScript has a special error for ! without : ("Declarations with definite
   # assignment assertions must also have type annotations.") but we parse it
   # so that the user can get this more useful error message.
-  NonNullAssertion _? (Colon Type)?:ct ->
+  _? NonNullAssertion _? (Colon Type)?:ct ->
```

Comment:
I don't necessarily want to allow whitespace before the non-null assertion since it will crowd out a possible future `!` binary op.